### PR TITLE
Added unit tests for .NET 7

### PR DESCRIPTION
**Pull request by hartmark**
Related to #451

### CODE REVIEW

File: src/Directory.Build.targets [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
   <Choose>
-    <When Condition="'$(TargetFramework)' == 'net462' OR '$(TargetFramework)' == 'netstandard2.0' OR '$(TargetFramework)' == 'net6.0'">
+    <When Condition="'$(TargetFramework)' == 'net462' OR '$(TargetFramework)' == 'netstandard2.0' OR '$(TargetFramework)' == 'net6.0' OR '$(TargetFramework)' == 'net7.0'">
```
Comment (abatishchev) [inline src/Directory.Build.targets:5]: I think some `|| NET7.0` are missing

File: tests/JWT.Tests.Common/JWT.Tests.Common.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net6.0;netstandard2.0;net462</TargetFrameworks>
+    <TargetFrameworks>net7.0;net6.0;netstandard2.0;net462</TargetFrameworks>
```
Comment (abatishchev) [inline tests/JWT.Tests.Common/JWT.Tests.Common.csproj:4]: My main question would be why to do this? In other words, what's the benefit of targeting net7 if we don't call any net7 specific API?

**Review by abatishchev — CHANGES_REQUESTED** [review]

File: src/Directory.Build.targets (hunk 1 above)
Comment (abatishchev) [reply src/Directory.Build.targets:5]: Also please update the Extension projects

**Review by abatishchev — CHANGES_REQUESTED** [review]

File: tests/JWT.Tests.Common/JWT.Tests.Common.csproj (hunk 2 above)
Comment (hartmark) [reply tests/JWT.Tests.Common/JWT.Tests.Common.csproj:4]: .net6 shouldn't have been needed too, as .netstandard2 should be enough. I'll check it out can be simplified

Comment (abatishchev) [reply tests/JWT.Tests.Common/JWT.Tests.Common.csproj:4]: The .NET Standard is dead, the recommendation by the .NET team is fo target certain "full" .NET version, such as 3.1 or 6. And that was my plan going forward - to add the LTS versions, i.e. skip 7.0 until really needed, snd later add 8.0

Comment (hartmark) [reply tests/JWT.Tests.Common/JWT.Tests.Common.csproj:4]: Alright, fair enough. I'll try remove .net7

**Comment by hartmark** [comment]
@abatishchev From your comment that the nugget should only target LTS versions of .net I have changed this PR to just add the Test-project of .net7 so we can be certain that everything still works for anyone using this nuget on .net7

**Review by abatishchev — DISMISSED** [review]

**Comment by abatishchev** [comment]
@hartmark can you please see why on earth the packages restore fails? :(

**Review by abatishchev — APPROVED** [review]

**Comment by hartmark** [comment]
Cool, nice cooperation 😎

Regarding nugget failure,  Microsoft has some issues earlier today, perhaps related. Good it's working now.

**Comment by abatishchev** [comment]
Cool, thanks for the contribution, appreciate it as always.

Not sure, if it was a network issue or just missing additional framework installed by the build pipeline.

**Pull request merged**